x/evm/types/response.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+import (
+	"strconv"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	proto "github.com/cosmos/gogoproto/proto"
+)
+
+// PatchTxResponses fills the evm tx index and log indexes in the tx result
+func PatchTxResponses(input []*abci.ExecTxResult) []*abci.ExecTxResult {
+	var (
+		txIndex  uint64
+		logIndex uint64
+	)
+	for _, res := range input {
+		// assume no error result in msg handler
+		if res.Code != 0 {
+			continue
+		}
+
+		var txMsgData sdk.TxMsgData
+		if err := proto.Unmarshal(res.Data, &txMsgData); err != nil {
```

Comment:
maybe can reuse [DecodeTxResponses](https://github.com/crypto-org-chain/ethermint/blob/e16d1048a8fe7ad401cbce95999114c5fa9ec771/x/evm/types/utils.go#L56) with error cb

Reply:
we need to keep all the responses here, only update some.

Reply:
and I hope we can avoid decoding/encoding here, just need more hacking into cosmos-sdk.